common/chat.cpp

```diff
@@ -426,7 +426,7 @@ struct common_chat_templates * common_chat_templates_init(
             LOG_ERR("%s: failed to parse tool use chat template (ignoring it): %s\n", __func__, e.what());
         }
     }
-    return tmpls;
+    return {tmpls, common_chat_templates_free};
```

Comment:
You can directly instantiate `common_chat_templates_ptr` instead of:

```c++
auto * tmpls = new common_chat_templates();
```

Reply:
oops indeed, thanks / done.